samples/boards/nrf/system_off/Kconfig

```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
```

Comment:
Forgive my ignorance, but does the build system now include a `Kconfig` file automatically if present in the root folder of the app?

Reply:
I don't know whether that's now done automatically.  Existing samples use this idiom.

Reply:
@tejlmand could you clarify this?

Reply:
If this idiom is no longer necessary, there should be a PR to update the Zephyr mainline to remove it, so people using existing code as examples don't get tasked to change things unrelated to the problem they're solving.  IMO.

Reply:
@carlescufi this is a known and documented feature:
https://docs.zephyrproject.org/latest/application/index.html#application-cmakelists-txt
> If your application has its own kernel configuration options, create a Kconfig file in the same directory as your application’s CMakeLists.txt.

But I would like to know why we want to restrict the `APP_RETENTION` to only this sample and not a general setting.
Bit maybe that's a future iteration where the code itself could be made available through a lib.

Reply:
> @carlescufi this is a known and documented feature:

Yes, that's been there "forever".   I though the question was about having to include the "real" one at the bottom of the application one, which is something the build system might work to avoid.

> But I would like to know why we want to restrict the `APP_RETENTION` to only this sample and not a general setting.
> Bit maybe that's a future iteration where the code itself could be made available through a lib.

Yes, until nrfx provides the core functionality of controlling retention based on addresses rather than sectors, and Zephyr can provide a convenience wrapper for Nordic to actually solve #31087,  *and* we look into how retention can be supported on multiple SOCs: it's application-specific.

Reply:
> I though the question was about having to include the "real" one at the bottom of the application one, which is something the build system might work to avoid.

Good idea. That should be possible, but are we sure there are not any special use cases where a sample might want to load the Kconfig as the last thing ?
Like loading it a custom place, for special layout handling.

Not to mention the work that might be needed outside Zephyr itself to adjust, like here:
https://github.com/zephyrproject-rtos/mcuboot/blob/3fc59410b633a6d83bbb534e43aac43160f9bd32/boot/zephyr/Kconfig#L582

Reply:
> are we sure there are not any special use cases where a sample might want to load the Kconfig somewhere else, and not as the last thing ?

No, which is why I don't think it's a good idea.  The existing solution works fine.